indexing-service/src/test/java/io/druid/indexing/overlord/ImmutableWorkerInfoTest.java

```diff
@@ -0,0 +1,52 @@
+/*
+ * Licensed to Metamarkets Group Inc. (Metamarkets) under one
+ * or more contributor license agreements. See the NOTICE file
+ * distributed with this work for additional information
+ * regarding copyright ownership. Metamarkets licenses this file
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License. You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied. See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+
+
+package io.druid.indexing.overlord;
+
+import com.fasterxml.jackson.databind.ObjectMapper;
+import com.google.common.collect.ImmutableSet;
+import io.druid.indexing.worker.Worker;
+import io.druid.jackson.DefaultObjectMapper;
+import junit.framework.Assert;
+import org.joda.time.DateTime;
+import org.junit.Test;
+
+public class ImmutableWorkerInfoTest
+{
+  @Test
+  public void testSerde() throws Exception
+  {
+    ImmutableWorkerInfo workerInfo = new ImmutableWorkerInfo(
+        new Worker(
+            "testWorker", "192.0.0.1", 10, "v1"
+        ),
+        2,
+        ImmutableSet.of("grp1", "grp2"),
+        ImmutableSet.of("task1", "task2"),
+        new DateTime("2015-01-01T01:01:01Z")
+    );
+    ObjectMapper mapper = new DefaultObjectMapper();
+    final ImmutableWorkerInfo serde = mapper.readValue(
+        mapper.writeValueAsString(workerInfo),
+        ImmutableWorkerInfo.class
+    );
+    Assert.assertEquals(workerInfo, serde);
```

Comment:
Since you're relying on the equals functionality, can you add tests for `equals`?